subsys/storage/storage_area/storage_area_store.c

```diff
@@ -0,0 +1,1155 @@
+	const size_t flen = SAS_ALIGNUP(cksize, align) - cksize;
+	uint8_t fill[align];
+	struct storage_area_iovec wr[2] = {
+		{
+			.data = store->sector_cookie,
+			.len = cksize,
+		},
+		{
+			.data = fill,
+			.len = flen,
+		},
+	};
+	int rc;
+
+	memset(fill, SAS_FILLVAL, sizeof(fill));
+	rc = storage_area_writev(store->area, wroff, wr, 2U);
+	if (rc != 0) {
+		goto end;
+	}
+
+	store->data->loc = cksize + flen;
+end:
+	if (rc != 0) {
+		LOG_DBG("add cookie failed for sector %x", store->data->sector);
+	}
+
+	return rc;
+}
+
+static int store_get_sector_cookie(const struct storage_area_store *store,
+				   uint16_t sector, void *cookie, size_t cksz)
+{
+	const sa_off_t rdoff = sector * store->sector_size;
+	struct storage_area_iovec rd = {
+		.data = cookie,
+		.len = SAS_MIN(cksz, store->sector_cookie_size),
+	};
+
+	return storage_area_readv(store->area, rdoff, &rd, 1U);
+}
+
+static int store_fill_sector(const struct storage_area_store *store)
+{
+	struct storage_area_store_data *data = store->data;
+	const struct storage_area *area = store->area;
+	const size_t secpos = data->sector * store->sector_size;
+	uint8_t buf[SAS_MAX(SAS_MINBUFSIZE, area->write_size)];
+	struct storage_area_iovec wr = {
+		.data = buf,
+	};
+	int rc = 0;
+
+	memset(buf, SAS_FILLVAL, sizeof(buf));
+	while (data->loc < store->sector_size) {
+		sa_off_t wroff = secpos + data->loc;
+
+		wr.len = SAS_MIN(sizeof(buf), store->sector_size - data->loc);
+		rc = storage_area_writev(area, wroff, &wr, 1U);
+		if (rc != 0) {
+			break;
+		}
+
+		data->loc += wr.len;
+	}
+
+	if (rc != 0) {
+		LOG_DBG("failed to fill sector %d", data->sector);
+	}
+
+	return rc;
+}
+
+static int store_erase_block(const struct storage_area_store *store)
+{
+	const struct storage_area *area = store->area;
+	const size_t erase_size = area->erase_size;
+	struct storage_area_store_data *data = store->data;
+	int rc = 0;
+
+	if ((data->sector * store->sector_size) % erase_size != 0U) {
+		goto end;
+	}
+
+	size_t sblock = (data->sector * store->sector_size) / erase_size;
+	size_t bcnt = MAX(1U, store->sector_size / erase_size);
+
+	rc = storage_area_erase(area, sblock, bcnt);
+
+	if (rc != 0) {
+		LOG_DBG("erase failed at block %d", sblock);
+	}
+end:
+	return rc;
+}
+
+static int store_move_record(struct storage_area_record *record,
+			     const struct storage_area_store_compact_cb *cb)
+{
+	if ((cb == NULL) || (cb->move == NULL) || (!cb->move(record))) {
+		return 0;
+	}
+
+	if (!store_record_valid(record)) {
+		LOG_DBG("invalid record, skipping move");
+		return 0;
+	}
+
+	const struct storage_area_store *store = record->store;
+	struct storage_area_store_data *data = store->data;
+	const struct storage_area *area = store->area;
+	const size_t sector_size = store->sector_size;
+	const size_t align = area->write_size;
+	const struct storage_area_record dest = {
+		.store = record->store,
+		.sector = data->sector,
+		.loc = data->loc,
+		.size = record->size};
```

Comment:
Trailing commas in structs allow to add lines without the need to touch other lines if necessary.

```suggestion
		.size = record->size,
	};
```